test/Ocelot.UnitTests/Configuration/UpstreamTemplatePatternCreatorTests.cs

```diff
@@ -230,11 +231,31 @@ public void should_create_template_pattern_that_matches_query_string_with_multip
 
         this.Given(x => x.GivenTheFollowingFileRoute(fileRoute))
             .When(x => x.WhenICreateTheTemplatePattern())
-            .Then(x => x.ThenTheFollowingIsReturned($"^(?i)/api/subscriptions/[^/]+/updates(|\\?)unitId={MatchEverything}&productId={MatchEverything}$"))
+            .Then(x => x.ThenTheFollowingIsReturned($"^(?i)/api/subscriptions/[^/]+/updates(\\/|\\?|$)unitId={MatchEverything}&productId={MatchEverything}$"))
             .And(x => ThenThePriorityIs(1))
             .BDDfy();
     }
 
+    [Theory]
+    [InlineData(@"/api/v1/abc?{everything}", false)]
+    [InlineData(@"/api/v1/abc2/{everything}", true)]

```

Comment:
1. Please include traits with the IDs from the resolved issue.
2. I notice this is test data for #2132, but I recommend adding more unit tests for issues #2064 and #2065. This would enable us to close them with this PR as well.